src/android/UploadTask.java

```diff
@@ -129,20 +110,20 @@ public UploadTask(@NonNull Context context, @NonNull WorkerParameters workerPara
                     .build();
         }
 
-        httpClient.dispatcher().setMaxRequests(workerParams.getInputData().getInt(KEY_INPUT_CONFIG_CONCURRENT_DOWNLOADS, 2));
+        httpClient.dispatcher().setMaxRequests(nextPendingUpload.getOutputData().getInt(KEY_INPUT_CONFIG_CONCURRENT_DOWNLOADS, 2));
```

Comment:
And the concurrency should later come from the init parameters and not when given a file for upload.

@zafirskthelifehacker can you update the readme to mention these releases are not stable and users should implement retry mechanisms on their sides in case updates break pending uploads? The stable release would be the 2.x one.